src/main/java/com/ncc/aif/ValidateAIF.java

```diff
@@ -63,75 +75,304 @@ private static void loadModel(Model model, CharSource ontologySource) {
-        } else if (params.isPresent("kbToValidate")) {
-            filesToValidate = ImmutableList.of(params.getExistingFile("kbToValidate"));
-        } else {
-            throw new RuntimeException("Either kbToValidate or kbsToValidate must be specified");
+        // Convert the specified domain ontologies (if any) to CharSources.
+        Set<CharSource> domainOntologySources = new HashSet<>();
+        for (String source : domainOntologies) {
+            File file = new File(source);
+            domainOntologySources.add(Files.asCharSource(file, Charsets.UTF_8));
+        }
+
+        // Finally, try to create the validator, but fail if required elements can't be loaded/parsed.
+        ValidateAIF validator = null;
+        try {
+            validator = create(ldcFlag, programFlag, nistFlag, ImmutableSet.copyOf(domainOntologySources));
+        }
+        catch (RuntimeException rte) {
+            logger.error("Could not read/parse all domain ontologies or SHACL files...exiting.");
+            logger.error("--> " + rte.getLocalizedMessage());
+            System.exit(FILE_ERROR);
         }
 
+        // Collect the file(s) to be validated.
+        final List<File> filesToValidate = new ArrayList<>();
+        if (!validationFiles.isEmpty()) {
+            for (String file : validationFiles) {
+                if (file.endsWith(".ttl")) {
+                    filesToValidate.add(new File(file));
+                }
+                else {
+                    logger.warn("Skipping file without .ttl suffix: " + file);
+                }
+            }
+        }
+        else { // -d option
+            for (String dirName : validationDirs) {
+                File dir = new File(dirName);
+                if (!dir.exists()) {
+                    logger.warn("Skipping non-existent directory: " + dirName);
+                }
+                else if (dir.isDirectory()) {
+                    File[] files = dir.listFiles(pathname -> pathname.toString().endsWith(".ttl"));
+                    if (files != null) {
+                        filesToValidate.addAll(Arrays.asList(files));
+                    }
+                }
+                else {
+                    logger.warn("Skipping non-directory: " + dirName);
+                }
+            }
+        }
+
+        if (filesToValidate.isEmpty()) {
+            logger.error("No files with .ttl suffix were specified.  Use -h option for help.");
+            System.exit(FILE_ERROR);
+        }
+
+        // Display a summary of what we're going to do.
+        if (!validationFiles.isEmpty()) {
+            logger.info("-> Validating KB(s): " + filesToValidate);
+        }
+        else { // We'd have failed by now if there were no TTL files in the directory
+            // This would need to be addressed if we supported validating files in N directories.
+            logger.info("-> Validating all KBs (*.ttl) in directory: " + validationDirs);
```

Comment:
Is there a reason to treat this differently from the files specified on the command line? I think both could be equally large

Reply:
Well, we definitely don't want to list all ttl files in the directory-- there could be thousands.  I guess you could specify 1,000 filenames on the command line as part of -f, although off-hand I'd think you'd just stick them in a directory and use -d.

But maybe it's prudent not to list the files we'll be validating with -f?

Reply:
You definitely could specify 1000s of filenames if you were using wildcards.  Imagine something like:
-f */*.ttl

Reply:
You're both right.  I changed it so it only lists the files if there are <= 5 of them.  But there's always a line of output whether you specified files or directories.